src/nnet3bin/nnet3-add-recurrent-io-to-egs.cc

```diff
@@ -0,0 +1,166 @@
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "nnet3/nnet-example.h"
+#include "nnet3/nnet-example-utils.h"
+#include "nnet3/nnet-utils.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+        "Add recurrent IOs to examples according to the input/output changes of nnet\n"
```

Comment:
<pre><code> 
Modify the examples to reserve space for storing
the recurrent nodes' states during state-preserving
RNN training. Both the input and output would be modified.

See : nnet3-train for more information on how these modified examples are going to be used.
</code></pre>